x-pack/plugins/cross_cluster_replication/server/lib/follower_index_serialization.test.js

```diff
@@ -0,0 +1,191 @@
+      };
+
+      expect(deserializeShard(serializedShard)).toEqual(deserializedShard);
+    });
+  });
+
+  describe('deserializeFollowerIndex()', () => {
+    it('deserializes Elasticsearch follower index object', () => {
+      const serializedFollowerIndex = {
+        index: 'follower index name',
+        shards: [{
+          shard_id: 'shard 1',
+        }, {
+          shard_id: 'shard 2',
+        }],
+      };
+
+      const deserializedFollowerIndex = {
+        name: 'follower index name',
+        shards: [{
+          id: 'shard 1',
+          remoteCluster: undefined,
+          leaderIndex: undefined,
+          leaderGlobalCheckpoint: undefined,
+          leaderMaxSequenceNum: undefined,
+          followerGlobalCheckpoint: undefined,
+          followerMaxSequenceNum: undefined,
+          lastRequestedSequenceNum: undefined,
+          outstandingReadRequestsCount: undefined,
+          outstandingWriteRequestsCount: undefined,
+          writeBufferOperationsCount: undefined,
+          writeBufferSizeBytes: undefined,
+          followerMappingVersion: undefined,
+          followerSettingsVersion: undefined,
+          totalReadTimeMs: undefined,
+          totalReadRemoteExecTimeMs: undefined,
+          successfulReadRequestCount: undefined,
+          failedReadRequestsCount: undefined,
+          operationsReadCount: undefined,
+          bytesReadCount: undefined,
+          totalWriteTimeMs: undefined,
+          successfulWriteRequestsCount: undefined,
+          failedWriteRequestsCount: undefined,
+          operationsWrittenCount: undefined,
+          readExceptions: undefined,
+          timeSinceLastReadMs: undefined,
+        }, {
+          id: 'shard 2',
+          remoteCluster: undefined,
+          leaderIndex: undefined,
+          leaderGlobalCheckpoint: undefined,
+          leaderMaxSequenceNum: undefined,
+          followerGlobalCheckpoint: undefined,
+          followerMaxSequenceNum: undefined,
+          lastRequestedSequenceNum: undefined,
+          outstandingReadRequestsCount: undefined,
+          outstandingWriteRequestsCount: undefined,
+          writeBufferOperationsCount: undefined,
+          writeBufferSizeBytes: undefined,
+          followerMappingVersion: undefined,
+          followerSettingsVersion: undefined,
+          totalReadTimeMs: undefined,
+          totalReadRemoteExecTimeMs: undefined,
+          successfulReadRequestCount: undefined,
+          failedReadRequestsCount: undefined,
+          operationsReadCount: undefined,
+          bytesReadCount: undefined,
+          totalWriteTimeMs: undefined,
+          successfulWriteRequestsCount: undefined,
+          failedWriteRequestsCount: undefined,
+          operationsWrittenCount: undefined,
+          readExceptions: undefined,
+          timeSinceLastReadMs: undefined,
+        }],
+      };
+
+      expect(deserializeFollowerIndex(serializedFollowerIndex)).toEqual(deserializedFollowerIndex);
```

Comment:
I think jest snapshots are the best for this type of tests. No need to set the `undefined` to all property. Just make sure the snapshot generated is correct. Much smaller test 😊